pkg/api/convertertoapi.go

```diff
@@ -591,12 +597,19 @@ func convertVLabsOrchestratorProfile(vp *vlabs.Properties, api *OrchestratorProf
 			convertVLabsKubernetesConfig(vlabscs.KubernetesConfig, api.KubernetesConfig)
 		}
 		setVlabsKubernetesDefaults(vp, api)
+
+		//HACK : this validation should be done as part of the main validation, but deploy does it only after loading the container.
+		if err := vp.ValidateOrchestratorProfile(isUpdate); err != nil {
```

Comment:
if we can get the deploycmd to run with `validation` set to true just like generate, then we can drop this hack.